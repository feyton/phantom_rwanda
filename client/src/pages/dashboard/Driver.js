import React from 'react';

function Driver() {
  return (
    <div>
      Drivers
      <span className="bg-get" />
      <span className="bg-create" />
      <span className="bg-update" />
      <span className="bg-delete" />
    </div>
  );
}

export default Driver;
