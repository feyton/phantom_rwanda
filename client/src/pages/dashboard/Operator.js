import React from 'react';

function Operator() {
  return (
    <div className="flex flex-row h-full w-full items-center justify-center">
      <h1 className="text-3xl font-bold">The operators page</h1>
    </div>
  );
}

export default Operator;
