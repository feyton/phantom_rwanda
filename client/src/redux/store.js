import { configureStore } from '@reduxjs/toolkit';
import { authReducer } from './reducers/authReducer.js';
import { busMovementReducer } from './reducers/busMovementReducer.js';
import { counterReducer } from './reducers/counterReducer.js';
import { dashboardReducer } from './reducers/dashboardReducer.js';
import { managementReducer } from './reducers/managementReducer.js';
import { profileReducer } from './reducers/profileReducer.js';
import { simulateReducer } from './reducers/simulateReducer.js';
import { socketReducer } from './reducers/socketReducer.js';
import { updateProfileReducer } from './reducers/updateProfileReducer.js';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    profileUpdates: updateProfileReducer,
    profileInfo: profileReducer,
    auth: authReducer,
    management: managementReducer,
    dashboard: dashboardReducer,
    socket: socketReducer,
    simulate: simulateReducer,
    movement: busMovementReducer
  }
});
