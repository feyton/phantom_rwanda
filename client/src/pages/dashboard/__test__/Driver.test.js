import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { store } from '../../../redux/store.js';
import Driver from '../Driver.js';
import Operator from '../Operator.js';

describe('Registration', () => {
  describe('Operators', () => {
    it('Should render the operator form', () => {
      const elem = renderer
        .create(
          <BrowserRouter>
            <Driver />
          </BrowserRouter>
        )
        .toJSON();
      expect(elem).toMatchSnapshot();
    });
    it('Should render the driver form', () => {
      const elem = renderer
        .create(
          <Provider store={store}>
            <BrowserRouter>
              <Operator />
            </BrowserRouter>
          </Provider>
        )
        .toJSON();
      expect(elem).toMatchSnapshot();
    });
  });
});
