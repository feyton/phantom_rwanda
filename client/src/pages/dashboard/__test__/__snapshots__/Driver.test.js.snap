// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Operators Should render the driver form 1`] = `
<div
  className="flex flex-row h-full w-full items-center justify-center"
>
  <h1
    className="text-3xl font-bold"
  >
    The operators page
  </h1>
</div>
`;

exports[`Registration Operators Should render the operator form 1`] = `
<div>
  Drivers
  <span
    className="bg-get"
  />
  <span
    className="bg-create"
  />
  <span
    className="bg-update"
  />
  <span
    className="bg-delete"
  />
</div>
`;
